x/warp/contract.go

```diff
@@ -256,22 +260,41 @@ func sendWarpMessage(accessibleState contract.AccessibleState, caller common.Add
 	}
 
 	// Add a log to be handled if this action is finalized.
+	topics, data, err := PackSendWarpMessageEvent(
+		destinationChainID,
+		destinationAddress,
+		sourceAddress,
+		unsignedWarpMessage.Bytes(),
+	)
+	if err != nil {
+		return nil, remainingGas, err
+	}
 	accessibleState.GetStateDB().AddLog(
 		ContractAddress,
-		[]common.Hash{
-			WarpABI.Events["SendWarpMessage"].ID,
-			destinationChainID,
-			destinationAddress.Hash(),
-			sourceAddress.Hash(),
-		},
-		unsignedWarpMessage.Bytes(),
+		topics,
+		data,
 		accessibleState.GetBlockContext().Number().Uint64(),
 	)
 
 	// Return an empty output and the remaining gas
 	return []byte{}, remainingGas, nil
 }
 
+// PackSendWarpMessageEvent packs the given arguments into SendWarpMessage events including topics and data.
+func PackSendWarpMessageEvent(destinationChainID common.Hash, destinationAddress common.Address, sourceAddress common.Address, unsignedMessageBytes []byte) ([]common.Hash, []byte, error) {
+	return WarpABI.PackEvent("SendWarpMessage", destinationChainID, destinationAddress, sourceAddress, unsignedMessageBytes)
```

Comment:
Seems we will be removing dstchain/dstaddr, so we should rebase this PR after the other work is merged

Reply:
yea, we will add new topic as well.